src/server/task_manager/task_store.ts

```diff
@@ -162,10 +162,10 @@ export class TaskStore {
       throw new Error(`Unsupported task type "${taskInstance.taskType}".`);
     }
 
-    const body = rawSource(taskInstance);
-    const { task } = body;
+    const { id, type, task } = rawSource(taskInstance);
```

Comment:
If we ever change raw source to return a new property, we'll have to modify this to structure the new property. Maybe this should be `{ id, ...body}` or something?